Compile failures on release model create/update

An `npm install` followed by `npm test` results in the type errors below during compilation. The release model [create()](https://github.com/balena-io-modules/balena-compose/blob/159cdbad4f9c91edd1b9b537a62cdfa1e3dcf309/lib/release/models.ts#L117) and `update()` functions expect a generic `body` object, but the API [post()](https://github.com/balena-io-modules/pinejs-client-js/blob/2e6e4a82b1e3798561a86cf48dc9177bf6e7a8d6/src/index.ts#L1088) and `patch()` methods expect the body to be an AnyObject.

```
$ npm test

> @balena/compose@2.1.1 test /home/kbee/dev/balena-compose/test-repo
> npm run lint && ts-mocha --project ./tsconfig.test.json


> @balena/compose@2.1.1 lint /home/kbee/dev/balena-compose/test-repo
> npm run lint:lib && npm run lint:tests && tsc --noEmit


> @balena/compose@2.1.1 lint:lib /home/kbee/dev/balena-compose/test-repo
> balena-lint --typescript lib/ typings/

Warning: The 'no-floating-promises' rule requires type information.
0 errors and 0 warnings in 4 files

> @balena/compose@2.1.1 lint:tests /home/kbee/dev/balena-compose/test-repo
> balena-lint --typescript --tests test/

Warning: The 'no-floating-promises' rule requires type information.
0 errors and 0 warnings in 18 files
lib/release/models.ts:122:30 - error TS2322: Type 'U' is not assignable to type 'AnyObject | undefined'.
  Type 'U' is not assignable to type 'AnyObject'.

122  return api.post({ resource, body }).catch(wrapResponseError) as Promise<T>;
                                 ~~~~

  lib/release/models.ts:117:27
    117 export function create<T, U>(
                                  ~
    This type parameter might need an `extends AnyObject` constraint.
  lib/release/models.ts:117:27
    117 export function create<T, U>(
                                  ~
    This type parameter might need an `extends AnyObject | undefined` constraint.
  node_modules/pinejs-client-core/index.d.ts:231:5
    231     body?: AnyObject;
            ~~~~
    The expected type comes from property 'body' which is declared here on type 'Params'

lib/release/models.ts:131:35 - error TS2322: Type 'T' is not assignable to type 'AnyObject | undefined'.
  Type 'T' is not assignable to type 'AnyObject'.

131  return api.patch({ resource, id, body }).catch(wrapResponseError);
                                      ~~~~

  lib/release/models.ts:125:24
    125 export function update<T>(
                               ~
    This type parameter might need an `extends AnyObject` constraint.
  lib/release/models.ts:125:24
    125 export function update<T>(
                               ~
    This type parameter might need an `extends AnyObject | undefined` constraint.
  node_modules/pinejs-client-core/index.d.ts:231:5
    231     body?: AnyObject;
            ~~~~
    The expected type comes from property 'body' which is declared here on type 'Params'


Found 2 errors in the same file, starting at: lib/release/models.ts:122

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @balena/compose@2.1.1 lint: `npm run lint:lib && npm run lint:tests && tsc --noEmit`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @balena/compose@2.1.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbee/.npm/_logs/2022-09-03T11_10_58_831Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! @balena/compose@2.1.1 test: `npm run lint && ts-mocha --project ./tsconfig.test.json`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @balena/compose@2.1.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbee/.npm/_logs/2022-09-03T11_10_58_846Z-debug.log
```

--- a/lib/release/models.ts
+++ b/lib/release/models.ts
@@ -1,5 +1,9 @@
 import type { PinejsClientRequest } from 'pinejs-client-request';
-import type { Expand, Filter, ODataOptions } from 'pinejs-client-core';
+import type {
+	Expand,
+	Filter,
+	ODataOptions,
+} from 'pinejs-client-core';
 
 import type { Composition } from '../../lib/parse';
 

--- a/lib/release/models.ts
+++ b/lib/release/models.ts
@@ -95,7 +95,7 @@ export interface ReleaseImageModel extends ReleaseImageAttributesBase {
 
 // Helpers
 
-export function getOrCreate<T, U, V extends Filter>(
+export function getOrCreate<T, U extends {}, V extends Filter>(
 	api: PinejsClientRequest,
 	resource: string,
 	body: U,
@@ -114,7 +114,7 @@ export function getOrCreate<T, U, V extends Filter>(
 	}) as Promise<T>;
 }
 
-export function create<T, U>(
+export function create<T, U extends {}>(
 	api: PinejsClientRequest,
 	resource: string,
 	body: U,
@@ -122,7 +122,7 @@ export function create<T, U>(
 	return api.post({ resource, body }).catch(wrapResponseError) as Promise<T>;
 }
 
-export function update<T>(
+export function update<T extends {}>(
 	api: PinejsClientRequest,
 	resource: string,
 	id: number,

--- a/package.json
+++ b/package.json
@@ -72,7 +72,7 @@
     "docker-modem": "^3.0.3",
     "docker-progress": "^5.1.0",
     "dockerfile-ast": "^0.2.1",
-    "dockerode": "^3.3.1",
+    "dockerode": "3.3.3",
     "duplexify": "^4.1.2",
     "event-stream": "^4.0.1",
     "fp-ts": "^2.8.1",

